### ISSUE TITLE
[Bug]: Trigger field UI to JS interaction looks jumpy when the field is at the end of the property pane

### ISSUE THREAD

**Issue by ramsaptami** [appsmithorg/appsmith#24388]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Toggling a trigger field from UI to JS when it's the last field on the property pane - example in the slider widget, the property pane height adjustment interaction looks glitchy and is not a smooth transition.

Converting from JS back to UI looks fine

### Steps To Reproduce

1. Add category slider widget to canvas or any other widget where the `events` field is at the end
2. Set any action on its `onChange` event from the UI
3. Toggle to JS mode and observe the height adjustment interaction causes the field to first go lower than its intended position and then come back up causing it to look jumpy.

### Environment

Production

### Version

Cloud

**Comment by bharath31** [comment]
@ramsaptami could you share a loom video?

**Comment by bharath31** [comment]
@ramsaptami bumping this up ICYMI. I'm not able to reproduce this

**Comment by ramsaptami** [comment]
@bharath31 attached a video to the description. Just tried it with a checkbox widget + store value function and I can repro it successfully.

**Comment by ramsaptami** [comment]
@Druthi yep, it's very much still there. Same steps as the video attached. We can go through it if needed 

Essentially what's happening on toggle is that, since the trigger field is already at the bottom of the property pane and the scroll bar is at the end, on switching to JS mode, the scroll bar is reverting to the top and coming back down to show the updated field state which is causing this jumpy interaction.

**Comment by Druthi** [comment]
Moving this back to icebox as it is a very deep issue which drills down through components and not easy to debug and also not a priority right now. Will pick it up again
